presto-main/src/main/java/com/facebook/presto/operator/scalar/SequenceFunction.java

```diff
@@ -96,19 +110,53 @@ public static Block sequenceTimestampYearToMonth(
         return blockBuilder.build();
     }
 
+    @ScalarFunction("sequence")
+    @SqlType("array(date)")
+    public static Block sequenceTimestampYearToMonthForDate(
+            ConnectorSession session,
+            @SqlType(StandardTypes.DATE) long start,
+            @SqlType(StandardTypes.DATE) long stop,
+            @SqlType(StandardTypes.INTERVAL_YEAR_TO_MONTH) long step)
+    {
+        checkCommonInvalidConditions(start, stop, step);
+
+        int length = toIntExact(diffDate(session, MONTH, start, stop) / step + 1);
+        checkMaxEntryCondition(length);
+
+        BlockBuilder blockBuilder = DATE.createBlockBuilder(new BlockBuilderStatus(), length);
+
+        int value = 0;
+        for (int i = 0; i < length; ++i) {
+            DATE.writeLong(blockBuilder, DateTimeOperators.datePlusIntervalYearToMonth(start, value));
+            value += step;
+        }
+
+        return blockBuilder.build();
+    }
+
     private static Block fixedWidthSequence(long start, long stop, long step, FixedWidthType type)
     {
-        checkCondition(step != 0, INVALID_FUNCTION_ARGUMENT, "step must not be zero");
-        checkCondition(step > 0 ? stop >= start : stop < start, INVALID_FUNCTION_ARGUMENT,
-                "sequence stop value should be greater than or equal to start value if step is greater than zero otherwise stop should be less than start");
+        checkCommonInvalidConditions(start, stop, step);
 
         int length = toIntExact((stop - start) / step + 1L);
-        checkCondition(length <= MAX_RESULT_ENTRIES, INVALID_FUNCTION_ARGUMENT, "result of sequence function must not have more than 10000 entries");
+        checkMaxEntryCondition(length);
 
         BlockBuilder blockBuilder = type.createBlockBuilder(new BlockBuilderStatus(), length);
         for (long i = 0, value = start; i < length; ++i, value += step) {
             type.writeLong(blockBuilder, value);
         }
         return blockBuilder.build();
     }
+
+    private static void checkCommonInvalidConditions(long start, long stop, long step)
+    {
+        checkCondition(step != 0, INVALID_FUNCTION_ARGUMENT, "step must not be zero");
+        checkCondition(step > 0 ? stop >= start : stop < start, INVALID_FUNCTION_ARGUMENT,
```

Comment:
The original formatting is not consistent with the preferred style. Write it as

```
checkCondition(
        step > 0 ? stop >= start : stop < start,
        INVALID_FUNCTION_ARGUMENT,
        "sequence stop value should be greater than or equal to start value if step is greater than zero otherwise stop should be less than start");
```

We prefer putting each argument (including the first one) on its own line.

Reply:
@ngqtien Have a separate commit that change "<" to "<=".